File: https:/github.com/rajesh-kayal-dev/Leetcode.git/48-RotateImage/48-RotateImage.java
// Last updated: 6/15/2026, 10:02:58 AM
class Solution {
    public void rotate(int[][] matrix) {
        swap(matrix);
        transpose(matrix);
    }

    public void swap(int[][] matrix) {
        int start =0;
        int end = matrix.length-1;

        while(start < end){
            int[] temp = matrix[start];
            matrix[start]=matrix[end];
            matrix[end]=temp;
            start++;
            end--;
        }
        
    }
    public void transpose(int[][] matrix) {
        for(int i = 0; i<matrix.length -1; i++){
            for(int j =i+1; j<matrix.length; j++){
                int temp=matrix[i][j];
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
    }

}

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/48-RotateImage/48-RotateImage.js
// Last updated: 7/25/2026, 10:59:43 AM
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var rotate = function(matrix) {
    
    //Transpose matrix

    for(let i = 0 ; i < matrix.length; i++){
        for(let j = i + 1 ; j < matrix.length; j++){
            [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]]
        }
    }

    //Reverse Matrix
    for(let row of matrix){
        row.reverse();
    }

};